Add $article_width variable for renderText option

DhruvaSambrani opened the pull request:
Add runtime var to send the actual width of the article to renderText command by text replacement

This allows renderText to have $article_width anywhere in the
command which will be replaced by the width of the article display
window.

Code Changes-
- Add article_width variable to TextConverter
- Add function helper.h#replaceString
- Change TextConverter to replace instances of $article_width with val

Other Changes-
- Change doc/neix.1 to reflect changes in options

Fixes #21

DhruvaSambrani committed:
Add $article_width variable for renderText option
Add runtime var to send the actual width of the article to renderText command by text replacement

This allows renderText to have $article_width anywhere in the
command which will be replaced by the width of the article display
window.

Code Changes-
- Add article_width variable to TextConverter
- Add function helper.h#replaceString
- Change TextConverter to replace instances of $article_width with val

Other Changes-
- Change doc/neix.1 to reflect changes in options

DhruvaSambrani committed:
M  doc/neix.1

tomschwarz commented:
@DhruvaSambrani 
Thanks for your work!

But there are some build errors when the Unit-Tests are enabled.
Could you check that and fix?

DhruvaSambrani commented:
I have fixed the error by changing param order and adding a default value. Should I also add a test case for this?

DhruvaSambrani committed:
Fix tests broken due to changed api

Textconverter tests were broken. Now they are fixed with an additional
test case for article_width

DhruvaSambrani commented:
@tomschwarz please check

tomschwarz commented:
@DhruvaSambrani 
Thanks for your fix! I really appreciate your work!
